File: app/src/main/java/demo/victormunoz/githubusers/features/userDetails/UserDetailsFragment.java
package demo.victormunoz.githubusers.features.userDetails;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.widget.NestedScrollView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import demo.victormunoz.githubusers.R;
import demo.victormunoz.githubusers.model.entity.User;


@SuppressWarnings("WeakerAccess")
public class UserDetailsFragment extends Fragment {
    @BindView(R.id.nestedScroll)
    NestedScrollView nestedScrollView;
    @BindView(R.id.user_bio)
    TextView biography;
    @BindView(R.id.user_created_at)
    TextView joined;
    @BindView(R.id.user_company)
    TextView company;
    @BindView(R.id.user_followers)
    TextView followers;
    @BindView(R.id.user_following)
    TextView following;
    @BindView(R.id.user_gists)
    TextView gists;
    @BindView(R.id.user_repos)
    TextView repos;
    @BindView(R.id.user_hireable)
    TextView hireable;
    @BindView(R.id.user_location)
    TextView location;

    public UserDetailsFragment(){
    }

    public static UserDetailsFragment newInstance(){
        return new UserDetailsFragment();
    }

    @Override
    public android.view.View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
        android.view.View root = inflater.inflate(R.layout.fragment_user_detail, container, false);
        ButterKnife.bind(this, root);
        return root;
    }

    public void displayUserInfo(@NonNull final User user){
        nestedScrollView.setVisibility(View.VISIBLE);
        followers.setText(user.getFollowers());
        following.setText(user.getFollowing());
        gists.setText(user.getTotalGists());
        repos.setText(user.getTotalRepos());
        joined.setText(getResources().getString(R.string.joined_on, user.getJoinedIn()));
        if (user.getBiography() != null) {
            biography.setVisibility(View.VISIBLE);
            biography.setText(user.getBiography());
        }
        if (user.getCompanyName() != null) {
            company.setVisibility(View.VISIBLE);
            company.setText(user.getCompanyName());
        }
        if (user.isHireable()) {
            hireable.setVisibility(View.VISIBLE);
            hireable.setText(R.string.available_for_hire);
        }
        if (user.getLocation() != null) {
            location.setVisibility(View.VISIBLE);
            location.setText(user.getLocation());
        }
    }

}

File: app/src/main/java/demo/victormunoz/githubusers/features/userDetails/UserDetailsFragment.kt
package demo.victormunoz.githubusers.features.userDetails

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import demo.victormunoz.githubusers.R
import demo.victormunoz.githubusers.model.User
import kotlinx.android.synthetic.main.fragment_user_detail.*


class UserDetailsFragment : Fragment() {

    companion object {

        fun newInstance(): UserDetailsFragment {
            return UserDetailsFragment()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): android.view.View? {
        return inflater.inflate(R.layout.fragment_user_detail, container, false)

    }

    fun displayUserInfo(user: User) {
        nsv_root.visibility = View.VISIBLE
        tv_followers.text = user.followers
        tv_following.text = user.following
        tv_gists.text = user.totalGists
        tv_repos.text = user.totalRepos
        tv_created_at.text = resources.getString(R.string.joined_on, user.joinedIn)
        user.biography?.run {
            tv_bio.visibility = View.VISIBLE
            tv_bio.text = this
        }
        user.companyName?.run {
            tv_company.visibility = View.VISIBLE
            tv_company.text = this
        }
        user.location?.run {
            tv_location.visibility = View.VISIBLE
            tv_location.text = this
        }
        if (user.isHireable) {
            tv_hireable.visibility = View.VISIBLE
            tv_hireable.setText(R.string.available_for_hire)
        }

    }

}
